### ISSUE TITLE
Update "What is a Grain Client?" doc page to 2.0 configuration API

### ISSUE THREAD

**Issue by sergeybykov** [dotnet/orleans#5081]
http://dotnet.github.io/orleans/Documentation/clusters_and_clients/index.html#initialization-of-grain-client is still referring to the legacy configuration API. We need to make it consistent with http://dotnet.github.io/orleans/Documentation/clusters_and_clients/configuration_guide/client_configuration.html. Otherwise, the page content is confusing.

**Comment by sergeybykov** [comment]
Still need to fix the example:
```csharp
var config = ClientConfiguration.LocalhostSilo();
var client = new ClientBuilder().UseConfiguration(config).Build();
await client.Connect();
```

**Comment by JillHeaden** [comment]
I changed these lines of code...

> var config = ClientConfiguration.LocalhostSilo();
> var client = new ClientBuilder().UseConfiguration(config).Build();
> await client.Connect();

… to this:

>             var client = new ClientBuilder()
>                 // Clustering information
>                 .Configure<ClusterOptions>(options =>
>                 {
>                     options.ClusterId = "my-first-cluster";
>                     options.ServiceId = "MyOrleansService";
>                 })
>                 // Clustering provider
>                 .UseAzureStorageClustering(options => options.ConnectionString = connectionString)
>                 // Application parts: just reference one of the grain interfaces that we use
>                 .ConfigureApplicationParts(parts => parts.AddApplicationPart(typeof(IValueGrain).Assembly))
>                 .Build();

Please let me know if this is not correct.

**Comment by JillHeaden** [comment]
Also, as you requested in your email, I removed the word "Awesome" from the sample code origin: http://dotnet.github.io/orleans/Documentation/clusters_and_clients/configuration_guide/client_configuration.html.

**Comment by sergeybykov** [comment]
There's one more instance to fix - https://github.com/dotnet/orleans/pull/5285/files#r244796965.

**Comment by JillHeaden** [comment]
Thank you! I have fixed it and pushed the change.

**Comment by sergeybykov** [comment]
Resolved via #5285.
